src/request/node.ts

```diff
@@ -144,17 +181,93 @@ export default function node<T>(url: string, options: NodeRequestOptions<T> = {}
 
 			// Redirection handling defaults to true in order to harmonise with the XHR provider, which will always
 			// follow redirects
-			// TODO: This redirect code is not 100% correct according to the RFC; needs to handle redirect loops and
-			// restrict/modify certain redirects
 			if (
 				response.statusCode >= 300 &&
-				response.statusCode < 400 &&
-				response.statusCode !== 304 &&
-				options.followRedirects !== false &&
-				nativeResponse.headers.location
+				response.statusCode < 400
 			) {
-				resolve(node(nativeResponse.headers.location, options));
-				return;
+				const redirectOptions = options.redirectOptions || {};
+				let newOptions = Object.create(options);
```

Comment:
I can't see where `newOptions` gets reassigned.  If it doesn't, it should be a `const`.